src/mlpack/tests/image_test.cpp

```diff
@@ -236,3 +299,34 @@ TEMPLATE_TEST_CASE("IdenticalResizeTest", "[ImageTest]", unsigned char, size_t,
   }
 }
 
+/**
+ * Test if we resize to the same original dimension we will get the same pixels
+ * and no modification to the image.  Try it with a few different types.
+ */
+TEMPLATE_TEST_CASE("IdenticalCropResizeTest", "[ImageTest]", unsigned char,
+    size_t, float, double)
+{
+  typedef TestType eT;
+
+  arma::Mat<eT> image;
+  data::ImageInfo info;
+  std::vector<std::string> files =
+      {"sheep_1.jpg", "sheep_2.jpg", "sheep_3.jpg", "sheep_4.jpg",
+       "sheep_5.jpg", "sheep_6.jpg", "sheep_7.jpg", "sheep_8.jpg", 
+       "sheep_9.jpg"};
+
+  for (size_t i = 0; i < files.size(); i++)
+  {
+    REQUIRE(data::Load(files.at(i), image, info, false) == true);
+    arma::Mat<eT> originalImage = image;
+    CropResizeImages(image, info, info.Width(), info.Height());
+    for (size_t i = 0; i < originalImage.n_rows; ++i)
+    {
+      for (size_t j = 0; j < originalImage.n_cols; ++j)
+      {
+        REQUIRE(originalImage.at(i, j) == image.at(i, j));
+      }
+    }
+  }
+}
```

Comment:
Do you think we could also add a test that if we crop/resize a 360x360 image to 180x180, that the pixels we get are the center 180x180 of the original image?  And the same for, e.g., 360x180 -> 180x180 and 180x360 -> 180x180.  I can help write that test if you want, just let me know.  But I am a little paranoid that we should actually check that we are getting the correct pixels as output, because it would be really easy to modify this function in the future in a way that causes these tests to pass, but actually garbles the pixel outputs.

Reply:
nope, so if the dimensions are equal then only resize will happen no need to crop the images
Crop only happens to keep the aspect ration without having a resolution problem,
The best as you said is to show before and after it will be much easier to understand.